src/rpc/rpcmapper.rs

```diff
@@ -348,107 +344,115 @@ mod tests {
                     buf
                 },
             };
-            let payload = UPayload::from_any(any);
-            Box::pin(std::future::ready(Ok(payload)))
+            let payload = any.into();
+            Ok(payload)
         }
     }
 
     struct ULinkHappyPath;
+
+    #[async_trait]
     impl RpcClient for ULinkHappyPath {
-        fn invoke_method(
-            topic: UUri,
-            payload: UPayload,
-            attributes: UAttributes,
-        ) -> Pin<Box<dyn Future<Output = RpcClientResult>>> {
+        async fn invoke_method(
+            _topic: UUri,
+            _payload: UPayload,
+            _attributes: UAttributes,
+        ) -> RpcClientResult {
             let payload = build_upayload_for_test();
-            Box::pin(std::future::ready(Ok(payload)))
+            Ok(payload)
         }
     }
 
     struct ULinkWithStatusCodeInsteadOfHappyPath;
+
+    #[async_trait]
     impl RpcClient for ULinkWithStatusCodeInsteadOfHappyPath {
-        fn invoke_method(
-            topic: UUri,
-            payload: UPayload,
-            attributes: UAttributes,
-        ) -> Pin<Box<dyn Future<Output = RpcClientResult>>> {
-            let status = ProtoStatus::from(UStatus::fail_with_msg_and_reason(
-                "boom",
-                UCode::InvalidArgument,
-            ));
-
-            let any = RpcMapper::pack_any(status.clone()).unwrap();
-            let payload = UPayload::from_any(any);
-
-            Box::pin(std::future::ready(Ok(payload)))
+        async fn invoke_method(
+            _topic: UUri,
+            _payload: UPayload,
+            _attributes: UAttributes,
+        ) -> RpcClientResult {
+            let status = UStatus::fail_with_code(UCode::InvalidArgument, "boom");
+
+            let any = RpcMapper::pack_any(status).unwrap();
```

Comment:
Since the function returns a `RpcClientResult`, what about removing the `unwrap()` that could cause panic and bubble up the error?

Reply:
done